### DISCUSSION TITLE
How to click an element with text

### DISCUSSION THREAD

**Discussion by maburrub** [seleniumbase/SeleniumBase#1085] [Q&A] [answered]
The framework allows you to click a link with text for example self.click_link_text(text) but this does not work if the element is not a link. Is there a way to do that by just providing the element text rather than a selector ?

**Comment by mdmintz** [comment, marked as answer]
@maburrub Use the ``:contains(text)`` selector.
E.g. ``self.click('button:contains("Submit")')``
